src/config/validator.rs

```diff
@@ -35,6 +35,10 @@ pub struct ValidatorConfig {
 
     /// Height at which to stop signing
     pub max_height: Option<tendermint::block::Height>,
+
+    /// Use Tendermint v0.33 handshake
+    #[serde(default = "handshake_default")]
+    pub v0_33_handshake: bool,
```

Comment:
Seems like we could have some better naming here? Perhaps just call the new handshake "V2", and use a handshake version number here rather than a boolean?

Reply:
The game plan is for Tendermint v0.34 to use GRPC so I think any versioning scheme is going to be confusing AF.

Reply:
What will the config look like in a GRPC world then?

I’m worried especially given that things could get complicated and hard-to-maintain.

Reply:
How should we disambiguate gRPC connections from Secret Connection V1 and V2 connections then? Will it use a different URI scheme?

I'm just worried because booleans are completely inextensible and would prefer something we can map to something like an enum in one way or another for future maintainability reasons, but if everything will drop Secret Connection V1 before the gRPC migration and we can safely delete all of the Secret Connection V1 codepaths before that happens, I guess it doesn't matter.

Reply:
How about using the Tendermint version as the configuration option?

Reply:
It looks like there is some breaking change in RPC as well.

I'm going to see if upgrading to `tendermint-rs 0.13` fixes it but that would pretty much abandon the idea of supporting multiple tendermint major versions in a single tmkms version completely.

Reply:
actually maybe not. Still confused.

Reply:
implemented @tarcieri 's suggestion